pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -156,7 +156,7 @@ set -e
 
 KEK_NAME=%s
 KEY_PATH=%s
-VAULT_DEFAULT_BACKEND=v1
+VAULT_DEFAULT_KV_VERS=v1
```

Comment:
Let's avoid changing the spec especially if it is cosmetic only, I do agree the change will be better though. But whoever is running Rook with KMS and upgrade Rook will get his/her OSD pod to restart for no reason.

Reply:
@leseb: need to change `VAULT_BACKEND` used in the script, because in rgw case the value for `VAULT_BACKEND` will set and it will be available as Environment variable for OSD pod
```
2021-01-12T12:15:32.7138654Z     Environment:
2021-01-12T12:15:32.7139098Z       VAULT_BACKEND_PATH:  rook
2021-01-12T12:15:32.7139548Z       VAULT_SKIP_VERIFY:   true
2021-01-12T12:15:32.7140178Z       KMS_PROVIDER:        vault
2021-01-12T12:15:32.7141086Z       VAULT_ADDR:          https://vault.default.svc.cluster.local:8200
2021-01-12T12:15:32.7141958Z       VAULT_BACKEND:       kv
2021-01-12T12:15:32.7142877Z       VAULT_TOKEN:         <set to the key 'token' in secret 'rook-vault-token'>  Optional: false
2021-01-12T12:15:32.7143498Z     Mounts:
```